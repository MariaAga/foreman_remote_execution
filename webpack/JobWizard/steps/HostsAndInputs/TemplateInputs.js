import React from 'react';
import PropTypes from 'prop-types';
import { translate as __ } from 'foremanReact/common/I18n';
import { TemplateInputsFields } from '../form/TemplateInputsFields';

export const TemplateInputs = ({ inputs, value, setValue }) => {
  if (inputs.length)
    return (
      <TemplateInputsFields inputs={inputs} value={value} setValue={setValue} />
    );
  return (
    <p className="gray-text">
      {__('There are no available input fields for the selected template.')}
    </p>
  );
};
TemplateInputs.propTypes = {
  inputs: PropTypes.array.isRequired,
  value: PropTypes.object,
  setValue: PropTypes.func.isRequired,
};

TemplateInputs.defaultProps = {
  value: {},
};
