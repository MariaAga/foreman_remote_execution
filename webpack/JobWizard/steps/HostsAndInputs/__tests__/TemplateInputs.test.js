import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import { TemplateInputs } from '../TemplateInputs';

const fixtures = {
  'renders with text props': {
    inputs: [
      {
        name: 'plain adv hidden',
        required: true,
        description: 'some Description',
        options: '',
        advanced: false,
        value_type: 'plain',
        resource_type: 'ansible_roles',
        default: 'Default val',
        hidden_value: true,
      },
    ],
    value: {},
    setValue: jest.fn(),
  },

  'renders without simple inputs': {
    inputs: [
      {
        name: 'plain adv hidden',
        required: true,
        description: 'some Description',
        options: '',
        advanced: true,
        value_type: 'plain',
        resource_type: 'ansible_roles',
        default: 'Default val',
        hidden_value: true,
      },],
    value: {},
    setValue: jest.fn(),
  },
};

describe('TemplateInputs', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(TemplateInputs, fixtures));
});
