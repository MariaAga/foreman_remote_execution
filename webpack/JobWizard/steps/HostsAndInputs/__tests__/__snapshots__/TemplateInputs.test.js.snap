// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateInputs rendering renders with text props 1`] = `
<Fragment>
  <FormGroup
    fieldId="plain adv hidden"
    isRequired={true}
    key="plain adv hidden"
    label="plain adv hidden"
    labelIcon={
      <Popover
        aria-label="help-text"
        bodyContent="some Description"
        id="undefined-help"
      >
        <button
          aria-label="open-help-tooltip-button"
          className="pf-c-form__group-label-help"
          onClick={[Function]}
          type="button"
        >
          <HelpIcon
            color="currentColor"
            noVerticalAlign={true}
            size="sm"
          />
        </button>
      </Popover>
    }
  >
    <TextArea
      className="masked-input"
      id="plain adv hidden"
      onChange={[Function]}
      required={true}
      rows={2}
    />
  </FormGroup>
</Fragment>
`;

exports[`TemplateInputs rendering renders without simple inputs 1`] = `
<p
  className="gray-text"
>
  There are no available input fields for the selected template.
</p>
`;
